addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxCubeBridgeDiscovery.java

```diff
@@ -85,6 +77,148 @@ public boolean isBackgroundDiscoveryEnabled() {
 		return true;
 	}
 
+	private synchronized void discoverCube() {
+		discoveryRunning = true;
+		Thread thread = new Thread("Sendbroadcast"){
+			public void run(){
+				sendDiscoveryMessage (MAXCUBE_DISCOVER_STRING);
+				try {
+					sleep(5000);
+				} catch (Exception e) {
+				}
+				discoveryRunning = false;
+				logger.trace( "Done sending broadcast discovery messages.");
+			}
+		};
+		thread.start();
+		receiveDiscoveryMessage();
```

Comment:
@marcelrv , I just noticed that this line makes the openHAB startup hanging for 8-10 seconds, if no cube is available in the network. Please note that the startBackgroundDiscovery() method should return immediately and not be long-running in any situation. Could you come up with some fix for this?